dask/dataframe/shuffle.py

```diff
@@ -25,6 +25,13 @@
 logger = logging.getLogger(__name__)
 
 
+def shuffle_layer_factory(cls, *args):
+    annotations = args[-1]
+    layer = cls(*args[:-1])
+    layer.annotations = annotations
+    return layer
```

Comment:
I'll change this so that annotations on the layer are initialised:

1. from the kwarg if explicitly provided.
2. derived from the config otherwise.

I don't think the two cases should interact.

Reply:
Implemented